contrib/prometheus-config-reloader/main.go

```diff
@@ -185,17 +187,21 @@ func (w *volumeWatcher) writeConfigMapFile(filename, content string) error {
 }
 
 func (w *volumeWatcher) ReloadPrometheus() error {
-	req, err := http.NewRequest("POST", w.cfg.reloadUrl, nil)
+	cmd := exec.Command("/bin/sh", "-c", "ps cax | grep '/bin/prometheus ' | grep -v grep | grep -o '^[ ]*[0-9]*'")
```

Comment:
`ps` is not really magic, we can just read the procfs and find the process ourselves without relying on `ps` and `grep`.